src/imars3d/ui/stages/dataloading.py

```diff
@@ -1,129 +1,114 @@
-    )
-    proj_root = param.String(doc="map data_root to string to avoid a FileSelector error")
-    recn_root = param.Path(
-        default=Path.home(),
-        doc="reconstruction results root, default should be proj_root/processed_data",
-    )
-    temp_root = param.Path(
-        default=Path.home() / Path("tmp"),
-        doc="intermedia results save location",
-    )
-    recn_name = param.String(
-        default="myrecon",
-        doc="reconstruction results folder name",
-    )
-    # data container
-    ct = param.Array(
-        doc="radiograph stack as numpy array",
-        precedence=-1,  # hide
+    # take over config from previous step
+    config_dict = param.Dict(
+        default={
+            "facility": "TBD",
+            "instrument": "TBD",
+            "ipts": 0,
+            "projectdir": "TBD",
+            "name": "TBD",
+            "workingdir": "TBD",
+            "outputdir": "TBD",
+            "steps": [],
+        },
+        doc="Configuration dictionary",
     )
-    ob = param.Array(
-        doc="open beam stack as numpy array",
-        precedence=-1,  # hide
-    )
-    dc = param.Array(
-        doc="dark current stack as numpy array",
-        precedence=-1,  # hide
-    )
-    omegas = param.Array(
-        doc="rotation angles in degress derived from tiff filename.",
-    )
-    # load data action
-    load_data_action = param.Action(lambda x: x.param.trigger("load_data_action"))
-    progress_bar = Tqdm(width=300, align="end")
+    # button to write filelist to json
+    update_config_action = param.Action(lambda x: x.param.trigger("update_config_action"))
+    save_config_to_disk = param.Action(lambda x: x.param.trigger("save_config_to_disk"))
 
-    @param.depends("data_root")
-    def _update_proj_root(self):
-        self.proj_root = str(self.data_root)
+    @param.depends("save_config_to_disk", watch=True)
+    def save_config_file(self):
+        wkdir = Path(self.config_dict["outputdir"])
+        config_filename = str(wkdir / f"{self.config_dict['name']}.json")
+        save_config(self.config_dict, config_filename)
+
+    @param.depends("update_config_action", watch=True)
+    def update_config(self):
+        # NOTE:
+        # the filter name get be get by
+        # left search bar entry: dataloader.radiograph_folder._selector._search[False].value
+        # right search bar entry: dataloader.radiograph_folder._selector._search[True].value
+        #
+        # NOTE:
+        # when a user is using the GUI to generate the list, it should include all the files,
+        # therefore no need for additional filtering by setting *_dc_fnmatch
+        #
+        # NOTE:
+        # it is better to hide the control of how many cores to use from user, hence using the
+        # default values for max_workers (by not setting one here).
+        load_step = {
+            "name": "load",
+            "version": 1,  # where does this version come from?
```

Comment:
we are not using `version`.
The `load` task should also define `"function"`, with its value being the function resolution when importing `load_step`, i.e., `"imars3d.backend.io.data.load_data"`

Reply:
`version` is removed, and `function` is added as suggested.